t/09_qesapdeployment.t

```diff
@@ -789,4 +789,96 @@ subtest '[qesap_prepare_env::qesap_create_aws_config] not solved template and va
     like $contents[0], qr/region = eu-central-1/, "Expected region eu-central-1 is in the config file";
 };
 
+subtest '[qesap_is_job_finished]' => sub {
+    my $qesap = Test::MockModule->new('qesapdeployment', no_auto => 1);
+    my @results = ();
+    $qesap->redefine(script_output => sub {
+            if ($_[0] =~ /100000/) { return "not json"; }
+            if ($_[0] =~ /200000/) { return "{\"job\":{\"state\":\"donaldduck\"}}"; }
+            if ($_[0] =~ /300000/) { return "{\"job\":{\"state\":\"running\"}}"; }
+    });
+
+    $qesap->redefine(get_required_var => sub { return ''; });
+    $qesap->redefine(record_info => sub { note(join(' ', 'RECORD_INFO -->', @_)); });
+
+    push @results, qesap_is_job_finished(100000);
+    push @results, qesap_is_job_finished(200000);
+    push @results, qesap_is_job_finished(300000);
+
+    my @expected_results = (0, 1, 0);
+
+    ok($results[0] == $expected_results[0], "Correct result if the openqa job status response isn't JSON");
+    ok($results[1] == $expected_results[1], "Correct result if the openqa job status response isn't 'running'");
+    ok($results[2] == $expected_results[2], "Correct result if the openqa job status response is 'running'");
+};
+
+subtest '[qesap_az_get_active_peerings]' => sub {
+    my $qesap = Test::MockModule->new('qesapdeployment', no_auto => 1);
+    my %results;
+    my @croak_calls;
+
+    $qesap->redefine(script_output => sub {
+            if ($_[0] =~ /myresourcegroup/ && $_[0] =~ /myvnetname/) {
+                return "vnet123456-vnet-other\nvnet789012-vnet-other\nvnet-nojobid-vnet-other";
+            }
+    });
+
+    $qesap->redefine(croak => sub { 
```

Comment:
look at `dies_ok` https://metacpan.org/pod/Test::Exception
look at some usage example of it in `t/09_qesapdeployment.t`

does this fit your needs? Can `dies_ok` be used in place of this croak redefine?

Reply:
I preferred this, in order to check against the actual output of the croak command (there are 2 cases where this may go wrong, one for each arg, with different output)

Reply:
```suggestion
subtest '[qesap_az_get_active_peerings] die for missing mandatory arguments' => sub {
    dies_ok { qesap_az_get_active_peerings(); } "Expected die if called without arguments";
    dies_ok { qesap_az_get_active_peerings(vnet => 'SEAWEED'); } "Expected die if called without rg";
    dies_ok { qesap_az_get_active_peerings(rg => 'CORAL'); } "Expected die if called without vnet";
};
```

Reply:
Ok, ok, I'll do it, you convinced me!